File: 2032-two-out-of-three/2032-two-out-of-three.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @param {number[]} nums3
 * @return {number[]}
 */
var twoOutOfThree = function(nums1, nums2, nums3) {
  const map = new Map()
  nums1.forEach((n, i) => {
    if (nums1.indexOf(n) === i) map.set(n, 1)
  })
  nums2.forEach((n, i) => {
    if (nums2.indexOf(n) === i) {
      if (map.has(n))
        map.set(n, map.get(n) + 1)
      else map.set(n, 1)
    }
  })
  nums3.forEach((n, i) => {
    if (nums3.indexOf(n) === i) {
      if (map.has(n))
        map.set(n, map.get(n) + 1)
      else map.set(n, 1)
    }
  })
  const ans = []
  for (let [k, v] of map) {
    if (v > 1) ans.push(k)
  }
  return ans
};

File: 2032-two-out-of-three/2032-two-out-of-three.go
func twoOutOfThree(nums1 []int, nums2 []int, nums3 []int) []int {
  flags := [101]int{}
  for _, v := range nums1 {
    flags[v] = 1
  }
  for _, v := range nums2 {
    flags[v] |= 2
  }
  for _, v := range nums3 {
    flags[v] |= 4
  }
  result := make([]int, 0, 100)
  for i, v := range flags {
    switch v {
    case 3:
      result = append(result, i)  
    case 6:
      result = append(result, i)  
    case 5:
      result = append(result, i)  
    case 7:
      result = append(result, i)  
    }
  }
  return result
}